solr/core/src/java/org/apache/solr/pkg/PackageLoader.java

```diff
@@ -78,29 +82,31 @@
   public PackageLoader(CoreContainer coreContainer) {
     this.coreContainer = coreContainer;
 
-    List<String> enabledPackages = StrUtils.splitSmart(localPkgsWhiteList, ',');
+    List<String> enabledPackages = StrUtils.splitSmart(enabledLocalPkgsList, ',');
     packageAPI = new PackageAPI(coreContainer, this);
 
     if (localPkgsDir != null && !enabledPackages.isEmpty()) {
-      loadLocalPackages(enabledPackages);
+      loadLocalPackages(localPkgsDir, enabledPackages);
+    }
+    if (repoPackagesEnabled) {
+      refreshPackageConf();
     }
-    if (enablePackages) refreshPackageConf();
   }
 
-  private void loadLocalPackages(List<String> enabledPackages) {
-    final Path packagesPath = new File(localPkgsDir.charAt(0)== File.separatorChar?
-            localPkgsDir :
-            coreContainer.getSolrHome()+ File.separator+ localPkgsDir).toPath();
+  private void loadLocalPackages(String localPkgsDir,  List<String> enabledPackages) {
+    final Path packagesPath = localPkgsDir.charAt(0) == File.separatorChar ?
+            Paths.get(localPkgsDir) :
+            Paths.get(coreContainer.getSolrHome()).resolve(localPkgsDir);
     log.info("Packages to be loaded from directory: {}", packagesPath);
 
-    if (!packagesPath.toFile().exists()) {
+    if (!Files.exists(packagesPath)) {
```

Comment:
Is there or will there be another way than sysProp to set the local packages dir? If it can be set by some api, it should be checked against allowPaths in cc.

Reply:
Sys prop is the only way to do it now. We should NOT have an API to do this because this also goes against the `immutable deployments` narrative . However we may, in the future,  use other startup mechanisms to configure the same (say `solr.xml`)